src/java.desktop/share/classes/javax/swing/AbstractButton.java

```diff
@@ -2146,9 +2146,9 @@ protected void init(String text, Icon icon) {
 
 
     /**
-     * This is overridden to return false if the current <code>Icon</code>'s
-     * <code>Image</code> is not equal to the
-     * passed in <code>Image</code> <code>img</code>.
+     * Returns <code>true</code> if the button icon for present button state
+     * is an instance of ImageIcon and the image it contains is same as the
+     * <code>Image</code>, <code>false</code> otherwise.
```

Comment:
we need to mention "same as the passed in Image" in both..

Reply:
added